tidy3d/components/data/data_array.py

```diff
@@ -80,6 +82,13 @@ def validate(cls, value):
 
         # loading from raw dict (usually from file)
         if isinstance(value, dict):
+            if value.get("tag") == "DATA_ITEM":
+                # Read from external file
+                with h5py.File(value["data_file"], "r") as f:
```

Comment:
probably we should 
```py
try:
    ...
except FileNotFoundError as e:
    ...
```
 here and raise a custom error 'data object "{}" stored in external file "{}" that is missing. Please include it and try again.' kind of thing.